src/LineBot/Actions/DateTimePickerAction.cs

```diff
@@ -0,0 +1,195 @@
+        Date,
+
+        /// <summary>
+        /// Picker Mode Time.
+        /// </summary>
+        Time,
+
+        /// <summary>
+        /// Picker Mode DateTime.
+        /// </summary>
+        DateTime
+    }
+
+    /// <summary>
+    /// Encapsulates a datetimepicker action.
+    /// </summary>
+    public sealed class DateTimePickerAction : IAction
+    {
+#pragma warning disable 0414 // Suppress value is never used.
+        [JsonProperty("type")]
+        [JsonConverter(typeof(EnumConverter<ActionType>))]
+        private readonly ActionType _type = ActionType.DateTimePicker;
+#pragma warning restore 0414
+
+        private string _label;
+        private string _data;
+        private DateTimePickerMode _mode;
+        private DateTime? _initial;
+        private DateTime? _min;
+        private DateTime? _max;
+
+        /// <summary>
+        /// Gets or sets the label.
+        /// <para>Max: 20 characters.</para>
+        /// </summary>
+        [JsonProperty("label")]
+        public string Label
+        {
+            get
+            {
+                return _label;
+            }
+
+            set
+            {
+                if (string.IsNullOrWhiteSpace(value))
+                    throw new InvalidOperationException("The label cannot be null or whitespace.");
+
+                if (value.Length > 20)
+                    throw new InvalidOperationException("The label cannot be longer than 20 characters.");
+
+                _label = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the string returned via webhook in the postback.data property of the <see cref="IPostback"/> event.
+        /// <para>Max: 300 characters.</para>
+        /// </summary>
+        [JsonProperty("data")]
+        public string Data
+        {
+            get
+            {
+                return _data;
+            }
+
+            set
+            {
+                if (string.IsNullOrWhiteSpace(value))
+                    throw new InvalidOperationException("The data cannot be null or whitespace.");
+
+                if (value.Length > 300)
+                    throw new InvalidOperationException("The data cannot be longer than 300 characters.");
+
+                _data = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the action mode for the datetimepicker.
+        /// </summary>
+        [JsonProperty("mode")]
+        [JsonConverter(typeof(EnumConverter<DateTimePickerMode>))]
+        public DateTimePickerMode Mode
+        {
+            get
+            {
+                return _mode;
+            }
+
+            set
+            {
+                _mode = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the initial datetime for the datetimepicker.
+        /// </summary>
+        [JsonProperty("initial")]
+        public DateTime? Initial
+        {
+            get
+            {
+                return _initial;
```

Comment:
We probably also need some checks to make sure this is between min and max and that means we also need to check if this value is set in the min/max properties.